fedimint-server/src/config/api.rs

```diff
@@ -9,13 +10,22 @@ use fedimint_core::db::Database;
 use fedimint_core::module::{
     api_endpoint, ApiAuth, ApiEndpoint, ApiEndpointContext, ApiError, ApiRequestErased,
 };
+use fedimint_core::PeerId;
+use jsonrpsee::core::Serialize;
 use jsonrpsee::server::{ServerBuilder, ServerHandle};
 use jsonrpsee::RpcModule;
+use serde::Deserialize;
+use tokio_rustls::rustls;
+use url::Url;
 
+use crate::config::{gen_cert_and_key, PeerServerParams};
 use crate::net::api::{attach_endpoints, HasApiContext, RpcHandlerCtx};
+use crate::net::client::WsAuthenticatedClient;
 
-/// Serves the config API endpoints
-pub struct ConfigApi {
+pub type Result<T> = std::result::Result<T, ApiError>;
```

Comment:
nit: maybe call this `ApiResult<T>` to make it distinguishable from the std `Result`? and then it's logical that the `ApiResult` returns an `ApiError`

Reply:
I don't think that's common, see e.g. `std::fmt::Result`, is there a Rust style guide to consult?

Reply:
I don't know about a style guide. With a quick search I found [`MathResult` in Rust by example](https://doc.rust-lang.org/rust-by-example/std/result.html) or [`LockResult` in std](https://doc.rust-lang.org/std/sync/type.LockResult.html). 
However, now while writing this, I see that these types always have a concrete type `T`; as does `std::fmt::Result`